recipes/ctpp_censustract_variables/build.sql

```diff
@@ -0,0 +1,197 @@
+                trans_public_rail numeric, trans_public_ferry numeric,
+                trans_bicycle numeric, trans_walk numeric, trans_taxi numeric,
+                trans_motorcycle numeric, trans_other numeric, trans_home numeric)
+    ),
+    MOE AS (
+        SELECT
+            a.geoid,
+            b.key as variable,
+            coalesce(b.value::numeric, 0)::integer as moe
+        FROM (
+            SELECT 
+                geoid, 
+                row_to_json(row) as _col
+            FROM (select * from _MOE) as row) a, 
+            json_each_text(_col) b
+        WHERE b.key != 'geoid'
+    ), 
+    _VAL AS (
+        SELECT 
+            a.geoid,
+            b.*,
+            
+            COALESCE(b.trans_auto_2, 0) +
+            COALESCE(b.trans_auto_3, 0) +
+            COALESCE(b.trans_auto_4, 0) +
+            COALESCE(b.trans_auto_5_or_6, 0) +
+            COALESCE(b.trans_auto_7_or_more, 0) 
+            as trans_auto_carpool_total, 
+
+            COALESCE(b.trans_auto_solo, 0) +
+            COALESCE(b.trans_auto_2, 0) +
+            COALESCE(b.trans_auto_3, 0) +
+            COALESCE(b.trans_auto_4, 0) +
+            COALESCE(b.trans_auto_5_or_6, 0) +
+            COALESCE(b.trans_auto_7_or_more, 0) 
+            as trans_auto_total,  
+
+            COALESCE(b.trans_public_bus, 0) +
+            COALESCE(b.trans_public_streetcar, 0) +
+            COALESCE(b.trans_public_subway, 0) +
+            COALESCE(b.trans_public_rail, 0) +
+            COALESCE(b.trans_public_ferry, 0) 
+            as trans_public_total
+
+        FROM RECODE a, 
+            -- pivot jsonb to columns (key -> field name, value -> field value), 
+            -- this step is needed because not all tracts have all modes of travel
+            jsonb_to_record(_val) as b(
+                trans_total numeric, trans_auto_solo numeric,
+                trans_auto_2 numeric, trans_auto_3 numeric,
+                trans_auto_4 numeric, trans_auto_5_or_6 numeric,
+                trans_auto_7_or_more numeric, trans_public_bus numeric,
+                trans_public_streetcar numeric, trans_public_subway numeric,
+                trans_public_rail numeric, trans_public_ferry numeric,
+                trans_bicycle numeric, trans_walk numeric, trans_taxi numeric,
+                trans_motorcycle numeric, trans_other numeric, trans_home numeric)
+    ), 
+    VAL AS (
+        SELECT
+            a.geoid,
+            b.key as variable,
+            coalesce(b.value::numeric, 0)::integer as "value"
+        FROM (
+            SELECT 
+                geoid, 
+                row_to_json(row) as _col
+            FROM (select * from _VAL) as row) a, 
+            json_each_text(_col) b
+        WHERE b.key != 'geoid'
+    )
+    SELECT
+        a.geoid, 
+        a.value::numeric::integer as value,
+        b.moe::numeric::integer as moe,
+        a.variable
+    FROM VAL a
+    FULL OUTER JOIN MOE b
```

Comment:
I don't think FULL OUTER is necessary. In what case would there be a mismatch between estimates and moes?